File: Equilibrium Point - GFG/equilibrium-point.py
# User function Template for python3
class Solution:
    # Complete this function
    
    #Function to find equilibrium point in the array.
    def equilibriumPoint(self,A, N):
        # Your code here
        total_sum = sum(A)
        curr_sum = 0
        half_sum = total_sum // 2
        
        for i in range(N):
            
            if curr_sum == (total_sum - A[i]) // 2:
                return i+1
            
            curr_sum += A[i]
        
        return -1


#{ 
 # Driver Code Starts
# Initial Template for Python 3

import math


def main():

    T = int(input())

    while(T > 0):

        N = int(input())

        A = [int(x) for x in input().strip().split()]
        ob=Solution()

        print(ob.equilibriumPoint(A, N))

        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends

File: Equilibrium Point - GFG/equilibrium-point.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;
import java.util.stream.*;

class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t =
            Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while (t-- > 0) {
            
            //taking input n
            int n = Integer.parseInt(br.readLine().trim());
            long arr[] = new long[n];
            String inputLine[] = br.readLine().trim().split(" ");
            
            //adding elements to the array
            for (int i = 0; i < n; i++) {
                arr[i] = Long.parseLong(inputLine[i]);
            }

            Solution ob = new Solution();
            
            //calling equilibriumPoint() function
            System.out.println(ob.equilibriumPoint(arr, n));
        }
    }
}
// } Driver Code Ends

class Solution {

    
    // a: input array
    // n: size of array
    // Function to find equilibrium point in the array.
 public static int equilibriumPoint(long arr[], int n) {        
        
        long ts = 0;
        long ls = 0;
         

        // loop for totalsum
        for(int i = 0;i<arr.length;i++){
            ts +=arr[i]; 
        }
        

         // loop for checking condition and calculating leftsum.
        for(int i = 0;i<arr.length;i++){
          if(ls==ts-ls-arr[i]){
              return i+1;
          }
            ls+=arr[i];
        }
        
     return -1;   
    }
}
